pinnwand/static/indent.js

```diff
@@ -0,0 +1,66 @@
+/// There should be a better way to handle the lexer
+/// I think :)
+
+
+function make_indent(event) {
+
+	var selector = document.querySelector("div.file-meta select[name='lexer']") // select the `select` under the file-meta div
+	var lexer = selector.options[selector.selectedIndex].text
+
+	// only indenting for python at the moment
+	if ( ! (lexer && lexer.toLowerCase().indexOf("python") == 0) ) {
+		return
+	}
+
+
+	var keyCode = event.keyCode || event.which;
+	if (keyCode == 9) { // check if a `tab` is pressed
+		event.preventDefault();
+		var start = this.selectionStart;
+		var end = this.selectionEnd;
+		
+		// set textarea value to: text before caret + tab + text after caret
+		val = this.value;
+		this.value = val.substring(0, start) + " ".repeat(4) + val.substring(end);
+		
+		// put caret at right position again
+		this.selectionStart =
+		this.selectionEnd = start + 4; // replace tab with 4 space, should make this configurable ?
+	} 
+	else if (keyCode == 13) { // check if an `enter` is pressed
+		setTimeout(function(that) {
```

Comment:
Is there a reason to perform this in a non-blocking way like this? Is it slow with a lot of text?

Reply:
Yeah, this was not necessary :)